patches/bunyan+1.8.15.patch

```diff
@@ -1,14 +1,43 @@
 diff --git a/node_modules/bunyan/lib/bunyan.js b/node_modules/bunyan/lib/bunyan.js
-index f988560..11af013 100644
+index f988560..a4cf69a 100644
 --- a/node_modules/bunyan/lib/bunyan.js
 +++ b/node_modules/bunyan/lib/bunyan.js
-@@ -76,7 +76,8 @@ if (runtimeEnv === 'browser') {
+@@ -63,7 +63,7 @@ if (!runtimeEnv) {
+ }
+
+
+-var os, fs, dtrace;
++var os, fs, pathModule, dtrace;
+ if (runtimeEnv === 'browser') {
+     os = {
+         hostname: function () {
+@@ -71,12 +71,15 @@ if (runtimeEnv === 'browser') {
+         }
+     };
+     fs = {};
++    pathModule = {};
+     dtrace = null;
+ } else {
      os = require('os');
      fs = require('fs');
++    pathModule = require('path');
      try {
 -        dtrace = require('dtrace-provider' + '');
 +        throw new Error('dtrace-provider is not available')
 +        // dtrace = require('dtrace-provider' + '');
      } catch (e) {
          dtrace = null;
      }
+@@ -1512,6 +1515,12 @@ RotatingFileStream.prototype.rotate = function rotate() {
+     }
+
+     function finish() {
++        if (!fs.existsSync(self.path)) {
++            var dirPath = pathModule.dirname(self.path);
++            if (!fs.existsSync(dirPath)) {
++                fs.mkdirSync(dirPath, { recursive: true });
++            }
++        }
```

Comment:
not sure why we need the line 35 here.
this could just be

```ts
+            var dirPath = pathModule.dirname(self.path);
+            if (!fs.existsSync(dirPath)) {
+                fs.mkdirSync(dirPath, { recursive: true });
+            }
```

couldn't it?

Reply:
Line 35 is so that we don't create a `dirPath` variable every time we rotate the log file (minor I know but good practice to check). The folder should only be created if the log file path cannot be reached since then we know for sure the directory doesn't exist.